src/coll_score/ucc_coll_score.c

```diff
@@ -733,18 +696,23 @@ static ucc_status_t ucc_coll_score_parse_str(const char *str,
     if (!ts_skip && (UCC_SCORE_INVALID != score_v || NULL != alg_id)) {
         /* Score provided but not coll_types/mem_types.
            This means: apply score to ALL coll_types/mem_types */
-        if (!ct)
+        if (!ct) {
             ct_n = UCC_COLL_TYPE_NUM;
-        if (!mt)
-            mt_n = UCC_MEMORY_TYPE_LAST;
-        if (!msg)
+        }
+        if (!mtypes) {
+            mtypes = -1;
```

Comment:
minor: maybe just leave a comment that "-1" means "ALL mtypes", ie full mask. Or even better, lets add a macro #define UCC_MEM_TYPE_MASK_FULL somewhere.

Reply:
added macro in src/utils/ucc_coll_utils.h